sound/soc/sof/topology.c

```diff
@@ -1871,14 +1874,61 @@ static int sof_dai_load(struct snd_soc_component *scomp, int index,
 	mutex_init(&spcm->mutex);
 	list_add(&spcm->list, &sdev->pcm_list);
 
-	return 0;
+	/* do we need to allocate playback PCM DMA pages */
+	if (!spcm->pcm.playback)
+		goto capture;
+
+	caps = &spcm->pcm.caps[stream];
+
+	/* allocate playback page table buffer */
+	ret = snd_dma_alloc_pages(SNDRV_DMA_TYPE_DEV, sdev->parent,
+				  PAGE_SIZE, &spcm->stream[stream].page_table);
+	if (ret < 0)
+		dev_err(sdev->dev, "error: can't alloc page table for %s %d\n",
+			caps->name, ret);
+
+	return ret;
```

Comment:
so we if we playback+capture we still return here? that looks like a mistake

Reply:
@plbossart yes, fixed now.